xdsl/parser/attribute_parser.py

```diff
@@ -1144,6 +1144,10 @@ def parse_optional_builtin_int_or_float_attr(
         if bool is not None:
             return bool
 
+        hexadecimal_token: Token | None = None
+        if self._current_token.text[:2] in ["0x", "0X"]:
+            hexadecimal_token = self._current_token
+
```

Comment:
I'm not sure this is the best way to do this, but it prevents losing the information that the token is a hexadecimal literal instead of a generic integer, until the attribute type is parsed.